hms-app/src/components/admin-tools/UserForm.tsx

```diff
@@ -0,0 +1,197 @@
+import { SkillPreview } from '../../common/types'
+import { getListOfSkills } from '../../actions/SkillActions'
+
+type IProps = {
+  contextID: string
+}
+
+const useStyles = createStyles((theme) => ({
+  card: {
+    backgroundColor:
+      theme.colorScheme === 'dark' ? theme.colors.dark[7] : theme.white,
+  },
+  section: {
+    borderBottom: `1px solid ${
+      theme.colorScheme === 'dark' ? theme.colors.dark[4] : theme.colors.gray[3]
+    }`,
+    paddingLeft: theme.spacing.md,
+    paddingRight: theme.spacing.md,
+    paddingBottom: theme.spacing.md,
+    textTransform: 'uppercase',
+    fontSize: theme.fontSizes.md,
+    fontWeight: 500,
+  },
+}))
+
+export default function UserForm(props: IProps) {
+  const { classes } = useStyles()
+  const { contextID } = props
+  const [isError, setIsError] = useState(false)
+  const [isLoading, setIsLoading] = useState(true)
+  const [availableSkills, setAvailableSkills] = useState({
+    skills: [] as SkillPreview[],
+  })
+  const [skills, setSkills] = useState<string[]>([])
+  const [user, setUser] = useState({
+    lastName: '',
+    firstName: '',
+    emailAddress: '',
+    roles: [],
+    imageUrl: '',
+  })
+
+  const loadSelectedUser = () => {
+    getUserDetails(contextID).then(
+      (data) => {
+        setIsError(false)
+        setIsLoading(false)
+        setUser({
+          lastName: data.lastName,
+          firstName: data.firstName,
+          emailAddress: data.emailAddress,
+          roles: data.roles,
+          imageUrl: data.imageUrl,
+        })
+      },
+      () => {
+        setIsError(true)
+        setIsLoading(false)
+      }
+    )
+  }
+
+  useEffect(() => {
+    loadSelectedUser()
+    loadAvailableSkills()
+    setIsLoading(true)
+  }, [])
+
+  function handleChange(event: React.ChangeEvent<HTMLTextAreaElement>) {
+    setUser((prevUserText) => ({
+      ...prevUserText,
+      [event.target.name]: event.target.value,
+    }))
+  }
+
+  function editThisUser(event: React.MouseEvent<HTMLButtonElement>) {
+    event.preventDefault()
+    showNotification({
+      id: 'user-load',
+      loading: true,
+      title: 'User is uploading',
+      message: 'this can take a second',
+      autoClose: false,
+      disallowClose: true,
+    })
+    editUser(contextID, user, skills).then((r) =>
+      setTimeout(() => {
+        updateNotification({
+          id: 'user-load',
+          color: 'teal',
+          title: 'User was added',
+          message: 'Notification will close in 2 seconds',
+          icon: <CheckIcon />,
+          autoClose: 2000,
+        })
+      }, 3000)
+    )
+  }
+
+  function submitIsEnabled(): boolean {
+    return !!user.firstName
+  }
+
+  const loadAvailableSkills = () => {
+    getListOfSkills().then((data) => {
+      setAvailableSkills({
+        ...availableSkills,
+        skills: data.skills,
+      })
+    })
+  }
+
+  const skillsList = availableSkills.skills.map((skill, index) => [
```

Comment:
this should then just be:
```suggestion
  const skillsList = availableSkills.map((skill, index) => [
```

Reply:
also I'm pretty sure that when you write these if you don't need the `index` you can just leave it out :thinking: